src/librustc_codegen_llvm/debuginfo/metadata.rs

```diff
@@ -761,14 +761,13 @@ fn hex_encode(data: &[u8]) -> String {
 
 pub fn file_metadata(
     cx: &CodegenCx<'ll, '_>,
-    file_name: &FileName,
+    source_file: &SourceFile,
```

Comment:
Heh, I think I did this (plus removing `defining_crate` and relying on the more accurate `source_file.is_imported()`) in one of my PRs that has been sitting around.